sentence-embedding/python-lib/dku_language_model/context_independent_language_model.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+from collections import Counter
+from gensim.models import KeyedVectors
+from sklearn.decomposition import TruncatedSVD
+from abstract_language_model import AbstractLanguageModel
+import logging
+logger = logging.getLogger(__name__)
+
+
+class ContextIndependentLanguageModel(AbstractLanguageModel):
+      
+    def __init__(self, model_path):
+        AbstractLanguageModel.__init__(self, model_path)
+        self.word2idx = None
+        self.embedding_matrix = None
+
+    def get_sentence_word_vectors(self, text):
+        indices = [self.word2idx[w] for w in text.split() if w in self.word2idx]
+        return self.embedding_matrix[indices]
+
+    def compute_average_embedding(self, text):
+        embeddings = self.get_sentence_word_vectors(text)
+        avg_embedding = np.mean(embeddings, axis=0)
+        return avg_embedding
+        
+    def get_sentence_embedding(self, texts):
+        cleaned_texts = map(self.clean_text, texts)
+        res = map(self.compute_average_embedding, cleaned_texts)
+        return res
+        
+    def get_weighted_sentence_word_vectors(self, text, weights):
+        #Check if sentence contains at least one token and return None if not
+        indices = [self.word2idx[w] for w in text.split() if w in self.word2idx]
+        embeddings = self.embedding_matrix[indices]
+        weights = [weights[w] for w in text.split() if w in self.word2idx]
```

Comment:
maybe run a single for loop, the two list comprehensions go through the same list

Reply:
yes this is what I'm intend to do, but I'm not sure I want to modify this much in this version.